File: 0975-range-sum-of-bst/0975-range-sum-of-bst.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} low
 * @param {number} high
 * @return {number}
 */
var rangeSumBST = function(root, low, high) {
    let values = [];
    let node = root;
    let stack =[];
    let s = 0;
    while (node || stack.length>0 ) {
        while (node) {
            stack.push(node);
            node = node.left;
        }

        node = stack.pop();
        values.push(node.val);
        node = node.right;
    }

    for (let j=0; j<=values.length; j++) {
        if (values[j]>=low && values[j]<=high) {
            s+=values[j];
        }
    }
    return s;
};

File: 0975-range-sum-of-bst/0975-range-sum-of-bst.py

# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        values = []
        node = root
        stack = []
        while node or stack:
            #go as left as possible
            while node:
                stack.append(node)
                node = node.left
            #visit node
            node = stack.pop()
            values.append(node.val)
            #move to the right subtree
            node = node.right
        return sum([i for i in values if i in range(low,high+1)])

                
        